### PR TITLE
fix: "use strict" should not trigger strict mode in es3.

### PR DESCRIPTION
**Pull request by aladdin-add**

### CODE REVIEW

File: tests/fixtures/ecma-version/3/strict/use-strict.result.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+                    }
+                },
+                "range": [
+                    183,
+                    195
+                ],
+                "value": "use strict",
+                "raw": "\"use strict\""
+            },
+            "directive": "use strict"
```
Comment (mdjermanovic) [inline tests/fixtures/ecma-version/3/strict/use-strict.result.js:53]: I'm not sure if this should be reported to Acorn as a possible bug. As far as I know, ES3 didn't have the concept of directive prologues, so maybe no nodes should be parsed as [Directive](https://github.com/estree/estree/blob/master/es5.md#directive) nodes in ES3 mode.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! 

I think we should merge this (and release it next week) regardless of https://github.com/eslint/espree/pull/547#discussion_r858617318 to ensure consistent behavior of Espree in regard to whether parsing produces syntax errors or not.

File: tests/fixtures/ecma-version/3/strict/use-strict.result.js (hunk 1 above)
Comment (nzakas) [reply tests/fixtures/ecma-version/3/strict/use-strict.result.js:53]: Agree. This seems like a bug in Acorn.

**Review by nzakas — COMMENTED** [review]
Can you also add a test for a function-level directive?

**Comment by aladdin-add** [comment]
@nzakas added in 7178c4f877afeb24af6472e9f15e039c8a7988e5.

**Review by nzakas — APPROVED** [review]
LGTM

**Pull request merged**